samples/spring-ai-agent/pom.xml

```diff
@@ -27,19 +27,74 @@
         <url/>
     </scm>
     <properties>
-        <java.version>17</java.version>
+        <java.version>21</java.version>
         <spring-ai.version>1.0.0</spring-ai.version>
+        <postgresql.version>42.7.5</postgresql.version>
+        <jib-maven-plugin.version>3.4.5</jib-maven-plugin.version>
+        <testcontainers.version>1.21.1</testcontainers.version>
     </properties>
     <dependencies>
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.ai</groupId>
+            <artifactId>spring-ai-starter-model-bedrock</artifactId>
+        </dependency>
         <dependency>
             <groupId>org.springframework.ai</groupId>
             <artifactId>spring-ai-starter-model-bedrock-converse</artifactId>
         </dependency>
 
+        <!-- RAG Dependencies -->
+        <dependency>
+            <groupId>org.springframework.ai</groupId>
+            <artifactId>spring-ai-vector-store</artifactId>
```

Comment:
There should be a starter package available so that we don't have to add them all manually: https://github.com/spring-projects/spring-ai/tree/main/spring-ai-spring-boot-starters/spring-ai-starter-vector-store-pgvector